tests/integration/testdata/buildcmd/terraform/zip_based_lambda_functions_local_backend/main.tf

```diff
@@ -494,4 +577,213 @@ resource "aws_lambda_function" "function4" {
         null_resource.build_hello_world_lambda_function, aws_s3_object.function4_code
     ]
 }
-## /* function4 connected to layer4
\ No newline at end of file
+## /* function4 connected to layer4
+
+## /* function5 connected to layer5
+resource "null_resource" "sam_metadata_aws_lambda_function5" {
+    triggers = {
+        resource_type = "ZIP_LAMBDA_FUNCTION"
+        original_source_code = local.hello_world_function_src_path
+        built_output_path = "${local.building_path}/${local.hello_world_artifact_file_name}"
+    }
+    depends_on = [
+        null_resource.build_hello_world_lambda_function
+    ]
+}
+
+resource "aws_s3_object" "function5_code" {
+  bucket = "existing_s3_bucket_name"
+  key    = "function5_code"
+  source = "${local.building_path}/${local.hello_world_artifact_file_name}"
+  depends_on = [
+    null_resource.build_hello_world_lambda_function
+  ]
+}
+
+resource "aws_lambda_function" "function5" {
+    s3_bucket = "existing_s3_bucket_name"
+    s3_key = "function5_code"
+    handler = "app.lambda_handler"
+    runtime = "python3.8"
+    function_name = "function5"
+    role = aws_iam_role.iam_for_lambda.arn
+    layers = [
+        aws_lambda_layer_version.layer5.arn,
+    ]
+    depends_on = [
+        null_resource.build_hello_world_lambda_function, aws_s3_object.function5_code
+    ]
+}
+## /* function5 connected to layer5
+
+
+## /* function6 connected to layer6
+resource "null_resource" "sam_metadata_aws_lambda_function6" {
+    triggers = {
+        resource_type = "ZIP_LAMBDA_FUNCTION"
+        original_source_code = local.hello_world_function_src_path
+        built_output_path = "${local.building_path}/${local.hello_world_artifact_file_name}"
+    }
+    depends_on = [
+        null_resource.build_hello_world_lambda_function
+    ]
+}
+
+resource "aws_lambda_function" "function6" {
+  filename = "${local.building_path}/${local.hello_world_artifact_file_name}"
+  handler = "app.lambda_handler"
+  runtime = "python3.8"
+  function_name = "function6"
+  role = aws_iam_role.iam_for_lambda.arn
+  layers = [
+      aws_lambda_layer_version.layer6.arn,
+  ]
+  depends_on = [
+      null_resource.build_hello_world_lambda_function
+  ]
+}
+## /* function6 connected to layer6
+
+# serverless.tf 3rd party module
+module "layer7" {
+  # this should be changed to `terraform-aws-modules/lambda/aws` when our change got merged and released`
```

Comment:
Another thing we should be tracking.

Reply:
this will be updated after the beta release